src/ray/protobuf/common.proto

```diff
@@ -256,6 +256,9 @@ message TaskSpec {
   int64 depth = 26;
   // Strategy about how to schedule this task.
   SchedulingStrategy scheduling_strategy = 27;
+  // A count of the number of times this task has executed so
+  // far. 0 means this is the first execution.
+  int64 execution_counter = 28;
```

Comment:
Nit: The semantics of this counter weren't obvious to me at first, since I first assumed that this was incremented at submission time ("number of executions, including the current one in progress"), but it actually isn't incremented until an execution run has finished (and failed). Would a better name for this be `num_past_executions` or something like that, so these semantics are more easily conveyed?

I'm also not super sure about this one, `execution_counter` is already a pretty good name for "number of executions that have already happened", so feel free to disregard if you think `execution_counter` is good enough.

Reply:
Hmm yeah not sure about it either, but I decided to just follow the naming convention we're already using for `parent_task_counter`.